### PR TITLE
Seek graph improvements

### PR DESCRIPTION
**Pull request by bhostetler18**
As proposed in #2050, this PR contains a number of improvements to the seek graph.

Notable changes:
- Improved resolution of seek graph
- Added shapes representing various challenge types (ranked, unranked, rengo)
- Moved challenge filters away from the bottom of the page, combining them with a legend explaining the symbols in the seek graph
- Made challenge filters affect the seek graph in addition to the challenge lists
- Added a challenge filter to hide/show rengo games
- Adjusted seek graph colors to match the current site theme
- Made the seek graph immediately respond to site theme changes
- Fixed a bug where the popover list would occasionally not show when mousing over a challenge in the bottom right of the seek graph

| Before  | After |
| ---------- | ---------- |
| old_seekgraph | new_seekgraph  |

### CODE REVIEW

**Comment by bhostetler18** [comment]
@anoek @benjaminpjones This is probably another good candidate for testing on the beta site first. There are a lot of code and visual changes, and some of the new behavior is opinionated enough to warrant community feedback.

**Comment by BHydden** [comment]
Can the key/legend/filters be minimized? That might be a good option to include if not.

**Comment by benjaminpjones** [comment]
I tried running this, but not seeing anything in the SeekGraph now.  I got an error about using keys in a map, but fixing it didn't help.

System: Firefox/Mac

Screen Shot 2022-10-21 at 8 44 16 PM

File: src/components/SeekGraph/SeekGraphLegend.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        ),
+    ];
+
+    return (
+        <div className="seek-graph-legend">
+            <Card>
+                <table>
+                    <tbody>
+                        {[0, 1, 2, 3, 4].map((i) => (
+                            <tr>
```
Comment (benjaminpjones) [inline src/components/SeekGraph/SeekGraphLegend.tsx:146]: Here's that key think I was talking about. ([React Docs: Lists and Keys](https://reactjs.org/docs/lists-and-keys.html))

```suggestion
                            <tr key={i}>
```

Comment (bhostetler18) [reply src/components/SeekGraph/SeekGraphLegend.tsx:146]: Thanks for the reminder! I’d actually meant to fix that but it completely slipped my mind.

Comment (bhostetler18) [reply src/components/SeekGraph/SeekGraphLegend.tsx:146]: The SeekGraph issue is probably entirely unrelated – it’s a different component than the legend. What happens when you inspect the area where it should show up?

Comment (bhostetler18) [reply src/components/SeekGraph/SeekGraphLegend.tsx]: Never mind, fixed. Firefox doesn’t support a certain font measure I was using, which caused a padding value to be NaN, which cascaded and made all rendering coordinates NaN…

**Comment by anoek** [comment]
What about putting the legend underneath the graph, and just squishing the graph vertically a little bit? I appologize for my GIMP skills but something kinda like this:
image

**Comment by bhostetler18** [comment]
I definitely considered it, but thought it might be unclear since the checkboxes also affect the challenge list. Happy to consider it further though, and I definitely agree that layout changes could be made.

Considering that people here and on the forums have asked for the option to hide the graph entirely – if the legend did go directly beneath the graph, should hiding the graph cause the legend to fill the full right card (on desktop at least)?

File: src/components/SeekGraph/SeekGraphLegend.tsx (hunk 1 above)
Comment (benjaminpjones) [reply src/components/SeekGraph/SeekGraphLegend.tsx]: yep the keys thing was def unrelated haha, just the first thing i saw in the console when I hit inspect.

Thanks for figuring out the Seek Graph issue - that was a strange root cause!

**Comment by anoek** [comment]
Hmm, I see what you're saying. 

On the game observation page we sorta solved this by having a filter icon that expands the filter options, though I'm not sure where such an icon would live on this page. 

Maybe for now, what about having those filters span the width of the automatch and seek graph, maybe 5 columns of two rows , and just have the groups wrap if there's not enough room (ie on mobile)?

**Comment by bhostetler18** [comment]
Sounds like a good idea! I’ll try it out.

I do wonder if it’s worth keeping the current layout on mobile, since it seems about as compact and nice-looking as possible:

74D9DADE-E06F-4FD6-9B4E-72EF29625B21

**Comment by anoek** [comment]
Yeah that sounds good

**Pull request merged**